sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java

```diff
@@ -381,12 +381,22 @@ static PathItem toPathItem(Path path) {
             parseDateOrNull(path.getLastModified()), path.getContentLength() == null ? 0 : path.getContentLength(),
             path.getGroup(), path.isDirectory() != null && path.isDirectory(), path.getName(), path.getOwner(),
             path.getPermissions(),
-            path.getCreationTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getCreationTime())),
-            path.getExpiryTime() == null ? null : fromWindowsFileTimeOrNull(Long.parseLong(path.getExpiryTime())));
+            parseWindowsFileTimeOrDateString(path.getCreationTime()),
+            parseWindowsFileTimeOrDateString(path.getExpiryTime()));
 
         return AccessorUtility.getPathItemAccessor().setPathItemProperties(pathItem, path.getEncryptionScope(), path.getEncryptionContext());
     }
 
+    static OffsetDateTime parseWindowsFileTimeOrDateString(String date) {
+        try {
+            // First try to parse as a long (Windows file time)
+            return fromWindowsFileTimeOrNull(Long.parseLong(date));
+        } catch (NumberFormatException ex) {
+            // If parsing as long fails, try to parse as a date string
+            return parseDateOrNull(date);
```

Comment:
I think this should have a try/catch as well where a clear `IllegalStateException` is thrown explaining the issue. This may help us in the future if this begins to return another format such as an ISO-8601 datestring (yyyy-MM-ddThh:mm:ss)